source/MaterialXCore/Types.cpp

```diff
@@ -85,6 +85,41 @@ template <> Matrix33 MatrixN<Matrix33, float, 3>::getAdjugate() const
         _arr[0][0]*_arr[1][1] - _arr[1][0]*_arr[0][1]);
 }
 
+Vector3 Matrix33::transform(const Vector3& rhs) const
+{
+    return Vector3(
+      _arr[0][0] * rhs[0] + _arr[0][1] * rhs[1] + _arr[0][2] * rhs[2],
+      _arr[1][0] * rhs[0] + _arr[1][1] * rhs[1] + _arr[1][2] * rhs[2],
+      _arr[2][0] * rhs[0] + _arr[2][1] * rhs[1] + _arr[2][2] * rhs[2]
+    );
+}
+
+Vector2 Matrix33::transformPoint(const Vector2& rhs) const
+{
+    Vector3 rhsVector3(rhs[0], rhs[1], 1.0f);
```

Comment:
As a naming convention for this variable (as well as similar variables below), what would you think of "rhs3"?  That seems less verbose without reducing clarity.